### PR TITLE
dnsimple terraform tls web application demo

### PR DESCRIPTION
**Pull request by AGS4NO**
Using Terraform to configure a web application with TLS using DNSimple.

### CODE REVIEW

File: terraform/tls-web-application/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+- terraform
+
+## Setup
+
+### Build Web Application
+
+Requires NodeJS + preferred package manager (examples using yarn)
+
+1. `cd react-app`
+2. `yarn && yarn build`
```
Comment (DXTimer) [inline terraform/tls-web-application/README.md:29]: What do you think about dropping the react-app in favour of a static page? My thinking is that we reduce the number of requirements and external dependencies by removing the need for Node, and we can still showcase the utility of the demo. WDYT?

File: terraform/tls-web-application/terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+resource "dnsimple_registered_domain" "domain" {
+  name = "${var.dnsimple_domain}"
+
+  contact_id            = 45600
+  auto_renew_enabled    = true
+  whois_privacy_enabled = true
+  dnssec_enabled        = false
+}
```
Comment (DXTimer) [inline terraform/tls-web-application/terraform/main.tf:44]: Have you considered omitting this resource from the config as it is not required, or maybe using the `dnsimple_domain` resource instead?

File: terraform/tls-web-application/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+See `terraform.tfvars.example` for examples.
+
+## Run Terraform
+
+1. `cd terraform`
+2. `terraform init`
+3. `terraform import dnsimple_registered_domain.domain {DNSIMPLE_REGISTERED_DOMAIN.TLD}` (Optional)
+4. `terraform plan`
+5. `terraform apply`
+
```
Comment (DXTimer) [inline terraform/tls-web-application/README.md:50]: I think perhaps a step is missing to add an entry in the hosts file for the domain on the machine.

File: terraform/tls-web-application/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+### Verify
```
Comment (DXTimer) [inline terraform/tls-web-application/README.md:51]: The ultimate verification step would be to visit the site and check the connection is encrypted. So maybe the link the user should visit highlighting to use the `8443` port?

File: terraform/tls-web-application/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+dnsimple_domain = "{DNSIMPLE_REGISTERED_DOMAIN.TLD}"
+```
+
+See `terraform.tfvars.example` for examples.
+
+## Run Terraform
+
+1. `cd terraform`
+2. `terraform init`
+3. `terraform import dnsimple_registered_domain.domain {DNSIMPLE_REGISTERED_DOMAIN.TLD}` (Optional)
```
Comment (DXTimer) [inline terraform/tls-web-application/README.md:47]: In the current setup this is not optional.

**Comment by DXTimer** [comment]
The demo worked great! It worked well and it did not require the provisioning of AWS resources but instead, everything was done locally ❤️. I left some feedback let me know what you think.

File: terraform/tls-web-application/README.md (hunk 1 above)
Comment (AGS4NO) [reply terraform/tls-web-application/README.md]: Agreed, I have moved the static assets into the `app` folder and removed the build requirement / node as a setup dependency.

File: terraform/tls-web-application/terraform/main.tf (hunk 2 above)
Comment (AGS4NO) [reply terraform/tls-web-application/terraform/main.tf]: Updated to use `dnsimple_domain`

File: terraform/tls-web-application/README.md (hunk 5 above)
Comment (AGS4NO) [reply terraform/tls-web-application/README.md]: Removed optional and switched the linked resource to `dnsimple_domain`

**Comment by AGS4NO** [comment]
I added in the following resource which calls a bash script to wait for a new certificate to be issued. The bash script will check every 30 seconds up to 10 minutes to ensure the certificate state is `issued` before creating the template files required for the container build.

```
resource "null_resource" "wait_for_certificate" {
  ...
}
```

**Review by DXTimer — APPROVED** [review]
Great work! I see that you also added a script to wait for the certificate to be issued. I am actually working on a PR to add retry behavior directly in the data source. But this is great as it is not blocking us! 
The dependency additions make this run much smoother 💯

**Pull request merged**